sktime/transformations/series/feature_selection.py

```diff
@@ -0,0 +1,196 @@
+        * "columns": Select features by given names.
+        * "none": Remove all columns by setting Z to None.
+        * "all": Select all given features.
+    regressor : sklearn-like regressor, optional, default=None.
+        Used as meta-model for the method "feature-importances". The given
+        regressor must have an attribute "feature_importances_". If None,
+        then a GradientBoostingRegressor(max_depth=5) is used.
+    n_columns : int, optional
+        Number of feautres (columns) to select. n_columns must be <=
+        number of X columns. Some methods require n_columns to be given.
+    random_state : int, RandomState instance or None, default=None
+        Used to set random_state of the default regressor and to
+        set random.seed() if method="random".
+    columns : list of str
+        A list of columns to select. If columns is given.
+
+    Attributes
+    ----------
+    columns_ : list of str
+        List of columns that have been selected as features.
+    regressor_ : sklearn-like regressor
+        Fitted regressor (meta-model).
+    n_columns_: int
+        Derived from number of features if n_columns is None, then
+        n_columns_ is calculated as int(math.ceil(Z.shape[1] / 2)). So taking
+        half of given features only as default.
+    feature_importances_ : dict or None
+        A dictionary with column name as key and feature imporatnce value as value.
+        The dict is sorted descending on value. This attribute is a dict if
+        method="feature-importances", else None.
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.feature_selection import FeatureSelection
+    >>> from sktime.datasets import load_longley
+    >>> y, X = load_longley()
+    >>> transformer = FeatureSelection(method="feature-importances", n_columns=3)
+    >>> Xt = transformer.fit_transform(X, y)
+    """
+
+    _tags = {
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": True,
+        "skip-inverse-transform": True,
+        "univariate-only": False,
+    }
+
+    def __init__(
+        self,
+        method="feature-importances",
+        n_columns=None,
+        regressor=None,
+        random_state=None,
+        columns=None,
+    ):
+        self.n_columns = n_columns
+        self.method = method
+        self.regressor = regressor
+        self.random_state = random_state
+        self.columns = columns
+
+        super(FeatureSelection, self).__init__()
+
+    def fit(self, Z, X=None):
+        """Fit the transformation on input series `Z`.
+
+        Parameters
+        ----------
+        Z : pd.DataFrame
+            A time series to apply the transformation on.
+
+        Returns
+        -------
+        self
+        """
+        Z = check_series(Z)
+        self.n_columns_ = self.n_columns
+        self.feature_importances_ = None
+
+        # multivariate Z
+        if not isinstance(Z, pd.Series):
```

Comment:
for readability it may be easier to read:
```suggestion
        if isinstance(Z, pd.Series):
            return self
        else:
```